### PR TITLE
[SPARK-25345][ML] Deprecate public APIs from ImageSchema

### PR DESCRIPTION
**Pull request by WeichenXu123**
## What changes were proposed in this pull request?

Deprecate public APIs from ImageSchema.

## How was this patch tested?

N/A

### CODE REVIEW

File: python/pyspark/ml/image.py [hunk 1]
```diff
@@ -20,6 +20,9 @@
 
     An attribute of this module that contains the instance of :class:`_ImageSchema`.
 
+.. note:: Deprecated in 2.4.0. Use `spark.read.format("image").load(path)` instead and
```
Comment (HyukjinKwon) [inline python/pyspark/ml/image.py]: FYI, that datasource wouldn't be able to replace `toNDArray` and `toImage` which require numpy. Might be better leave a note that it needs manual conversion with NumPy API for both.

File: mllib/src/main/scala/org/apache/spark/ml/image/ImageSchema.scala [hunk 2]
```diff
@@ -35,6 +35,8 @@ import org.apache.spark.sql.types._
  */
 @Experimental
 @Since("2.3.0")
+@deprecated("use `spark.read.format(\"image\").load(path)` and this `ImageSchema` will be " +
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/ml/image/ImageSchema.scala]: There are other methods defined under `ImageSchema` that are not covered by the image data source. So we shall only deprecate `readImages` and leave other public methods as experimental. Same for Python.

File: python/pyspark/ml/image.py [hunk 3]
```diff
@@ -207,6 +207,9 @@ def readImages(self, path, recursive=False, numPartitions=-1,
         .. note:: If sample ratio is less than 1, sampling uses a PathFilter that is efficient but
             potentially non-deterministic.
 
+        .. note:: Deprecated in 2.4.0. Use `spark.read.format("image").load(path)` instead and
```
Comment (holdenk) [inline python/pyspark/ml/image.py]: Would it make sense to issue a warning when people call this code path (e.g. `warnings.warn`)? We do it in some of the other Python deprecated APIs but not all.

Comment (WeichenXu123) [reply python/pyspark/ml/image.py]: added.

File: python/pyspark/ml/image.py [hunk 4]
```diff
@@ -30,6 +30,7 @@
 from pyspark import SparkContext
 from pyspark.sql.types import Row, _create_row, _parse_datatype_json_string
 from pyspark.sql import DataFrame, SparkSession
+import warnings
```
Comment (HyukjinKwon) [inline python/pyspark/ml/image.py]: Technically builtin package should be ordered above per PEP 8. I wonder why this is not caught.

```
import sys
import warnings
 
import numpy as np

from pyspark import SparkContext
from pyspark.sql.types import Row, _create_row, _parse_datatype_json_string
from pyspark.sql import DataFrame, SparkSession
```

File: python/pyspark/ml/image.py [hunk 5]
```diff
@@ -222,7 +226,8 @@ def readImages(self, path, recursive=False, numPartitions=-1,
 
         .. versionadded:: 2.3.0
         """
-
+        warnings.warn("`ImageSchema.readImage` is deprecated. " +
+                      "Use `spark.read.format(\"image\").load(path)` instead.")
```
Comment (HyukjinKwon) [inline python/pyspark/ml/image.py]: `warnings.warn(..., DeprecationWarning)`

**Review by HyukjinKwon — APPROVED** [review]

**Comment by mengxr** [comment]
@WeichenXu123 Could you address the comments?

**Comment by mengxr** [comment]
LGTM. Merged into master and branch-2.4. Thanks!

**Pull request closed without merging**